core/src/main/scala/scalaz/IList.scala

```diff
@@ -477,10 +477,17 @@ final case class ICons[A](head: A, tail: IList[A]) extends IList[A]
 object IList extends IListInstances {
   private[this] val nil: IList[Nothing] = INil()
 
+  // optimised versions of apply(A*)
+  @inline final def apply[A](a: A, b: A): IList[A] = a :: single(b)
+  @inline final def apply[A](a: A, b: A, c: A): IList[A] = a :: apply(b, c)
+  @inline final def apply[A](a: A, b: A, c: A, d: A): IList[A] = a :: apply(b, c, d)
+  @inline final def apply[A](a: A, b: A, c: A, d: A, e: A): IList[A] = a :: apply(b, c, d, e)
+  @inline final def apply[A](a: A, b: A, c: A, d: A, e: A, f: A): IList[A] = a :: apply(b, c, d, e, f)
```

Comment:
check if the `@inline` does anything?

Reply:
yeah it does, but only if you turn on the optimisations and whitelist `scalaz.*`